benchmark/src/main/java/com/zaxxer/nuprocess/benchmark/NuProcessBenchmark.java

```diff
@@ -0,0 +1,98 @@
+package com.zaxxer.nuprocess.benchmark;
+
+import com.zaxxer.nuprocess.NuAbstractProcessHandler;
+import com.zaxxer.nuprocess.NuProcess;
+import com.zaxxer.nuprocess.NuProcessBuilder;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.profile.GCProfiler;
+import org.openjdk.jmh.profile.HotspotThreadProfiler;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+import java.util.zip.CRC32;
+
+/**
+ * A simple performance benchmark which uses {@code cat} to write a file to {@code stdout}.
+ * <p>
+ * This benchmark requires a test file to use. One method to generate a test file is:
+ * <code><pre>
+ * head -c 250M /dev/urandom >/tmp/random.dat
+ * </pre></code>
+ * <p>
+ * The CRC32 for the file is also required to allow the benchmark to apply a spot check on the contents, after
+ * they've been pumped through {@code cat}, to verify that the contents were not corrupted during processing. A
+ * CRC32 checksum is used in preference to algorithms like SHA-256 because the speed of the hashing algorithm is
+ * not the focus of this benchmark.
+ * <p>
+ * After generating a test file, you can compute the CRC32 to update the benchmark by either running it once and
+ * checking the value in the error message, or running:
+ * <code><pre>
+ * crc32 /tmp/random.dat
+ * </pre></code>
+ * That will display the CRC32 checksum in hex, which can then be converted to decimal to match {@code CRC32}'s
+ * {@code getValue()}. (The calculator on most operating systems can convert hex to decimal.)
+ */
+public class NuProcessBenchmark {
```

Comment:
I wish that there was a way to make this a bit more self-contained. I had thought about just having it directly fork `head -c 250MB /dev/urandom`, but that relies on the system's ability to generate sufficient entropy. Otherwise it gets extremely slow, which would skew the benchmark. (It also makes it impossible to perform any sort of quick validation on the output to test for corruption, since the checksum would continuously change)